code/game/objects/items.dm

```diff
@@ -639,98 +643,110 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
-		if(block_flags & BLOCKING_NASTY)
-			if(istype(L, /mob/living/simple_animal))
-				var/mob/living/simple_animal/S = L
-				if(!S.hardattacks)
-					INVOKE_ASYNC(S, TYPE_PROC_REF(/atom, attackby), src, owner)
-					owner.visible_message(span_danger("[S] injures themselves on [owner]'s [src]!"))
-			else
-				INVOKE_ASYNC(L, TYPE_PROC_REF(/atom, attackby), src, owner)
-				owner.visible_message(span_danger("[L] injures themselves on [owner]'s [src]!"))
-	owner.apply_damage(attackforce, STAMINA, blockhand, block_power)
-	if((owner.getStaminaLoss() >= 35 && HAS_TRAIT(src, TRAIT_NODROP)) || (HAS_TRAIT(owner, TRAIT_NOLIMBDISABLE) && owner.getStaminaLoss() >= 30))//if you don't drop the item, you can't block for a few seconds
-		INVOKE_ASYNC(owner, TYPE_PROC_REF(/mob/living/carbon/human, blockbreak))
-	if(attackforce)
-		owner.changeNext_move(CLICK_CD_MELEE)
+
+		//Blocking gets a bonus against weapons that don't get their power from brute force, but the weight also doesn't matter
+		else if(!I.damtype == BRUTE)
+			attackforce = (attackforce * 0.8)
+
+		//When blocking a sharp weapon, the force conveyed is determined purely by its weight rather than its damage
+		else if(I.is_sharp())
+			attackforce = I.w_class * 2
+
+		//And if it's a blunt weapon, blocking takes the worst outcome between weight and damage bonuses
+		else
+			attackforce = max(I.w_class * 3, attackforce * 1.5)
+
+		//Is it a weapon especially adept at counterattacks? If so we roll for one
+		if(block_flags & BLOCKING_COUNTERATTACK && prob(50))
+			//is the item we blocked held by a mob, or was it thrown at us? We can't counter attack a thrown item.
+			if(isliving(hitby.loc))
+				var/mob/living/living_enemy = hitby.loc
+				INVOKE_ASYNC(living_enemy, TYPE_PROC_REF(/atom, attackby), src, owner)
+
+	//if it's not a weapon and it's not a projectile, we need to check for counterattacks and blocking_nasty
+	else if(attack_type == UNARMED_ATTACK && unarmed_mob && block_flags & (BLOCKING_NASTY|BLOCKING_COUNTERATTACK))
+		INVOKE_ASYNC(attacking_mob, TYPE_PROC_REF(/atom, attackby), src, owner)
+		owner.visible_message(span_danger("[attacking_mob] injures themselves on [owner]'s [src]!"))
+
+	//If this weapon is prone to knocking the opponent off balance, we want to delay their next attack and knock them down
+	if(block_flags & BLOCKING_UNBALANCE && prob(20) && attacking_mob)
+		attacking_mob.Knockdown(1 SECONDS)
+		attacking_mob.changeNext_move(CLICK_CD_MELEE * 2)
+
+	//If the attacker is a simple_animal we need to check if they are designed to be especially good against blocking
+	if(istype(hitby, /mob/living/simple_animal))
+		var/mob/living/simple_animal/simplemob = hitby
+		if(simplemob.hardattacks)
+			attackforce = attackforce * 5 //You can probably only block them once or twice at most because of stamina and/or shield damage
+
+	//We are ready to deal stamina damage to our owner
+	owner.apply_damage(min(attackforce, 60), STAMINA, blocked = block_power)
```

Comment:
I forgot to come back and improve this until I'd already posted this. I need to make certain this can't stamcrit